### ISSUE TITLE
Using CUDACHECK(cudaStreamGetCaptureInfo_v2(...)) breaks enhanced compatibility

### ISSUE THREAD

**Issue by cliffwoolley** [NVIDIA/nccl#496]
If we have an R460 driver (or earlier, back to 450.80.02), we should be able to still use versions of NCCL built with newer CUDA Toolkit 11.x via enhanced compatibility.

However, at this line: https://github.com/NVIDIA/nccl/blob/a46ea105830fbc3fbb222a00a82862df49cc9a9f/src/enqueue.cc#L802 , if we have an R460 driver, then we will get an error like this:

```
[1,0]<stdout>:8470dd07de42:174:239 [0] enqueue.cc:802 NCCL WARN Cuda failure 'API call is not supported in the installed CUDA driver'
[1,0]<stdout>:8470dd07de42:174:239 [0] misc/argcheck.cc:39 NCCL WARN Broadcast : invalid root 0 (root should be in the 0..-1 range)
[1,0]<stdout>:8470dd07de42:174:239 [0] init.cc:895 NCCL WARN Cuda failure 'invalid device ordinal'
```
Leading to the NCCL call failing with "Unhandled CUDA error".  

Technically we could have just skipped the CUDACHECK() guard there, and then the subsequent line `if (captureStatus == cudaStreamCaptureStatusActive) {` would evaluate to false, and comm->usingCudaGraph would remain false, and NCCL should proceed as it did prior to CUDA 11.3.

**Comment by kwen2501** [comment]
Thanks for reporting it.
NCCL does not yet support enhanced compatibility, but we will consider it on our roadmap.

**Comment by cliffwoolley** [comment]
This was effectively fixed in NCCL 2.9.9 via https://github.com/NVIDIA/nccl/commit/3fec2fa5eea34917f09e2282afc38863b454eae1#diff-adabec4bcb917d3813fc4640158f06f56ac041a8c3277efe7dd72ddaf2ea2e49R806 .  Closing.
